### PR TITLE
[Internal][SDK] Separate core connection

### PR DESCRIPTION
**Pull request by brynn-code**
# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

# All Promptflow Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes].**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](../CONTRIBUTING.md).**
- [ ] **Create an issue and link to the pull request to get dedicated review from promptflow team. Learn more: [suggested workflow](../CONTRIBUTING.md#suggested-workflow).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### Testing Guidelines
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: src/promptflow/promptflow/_sdk/entities/_connection.py [hunk 1]
```diff
@@ -287,13 +233,13 @@ def _get_scrubbed_secrets(self):
         return {key: val for key, val in self.secrets.items() if self._is_scrubbed_value(val)}
 
 
-class _StrongTypeConnection(_Connection):
+class _StrongTypeConnection(_CoreStrongTypeConnection, _Connection):
```
Comment (wangchao1230) [inline src/promptflow/promptflow/_sdk/entities/_connection.py:236]: entities.OpenAIConnection can be passed to place where accepts a core.OpenAIConnection, right?

File: src/promptflow/promptflow/_sdk/entities/_connection.py [hunk 2]
```diff
@@ -333,511 +279,81 @@ def _from_mt_rest_object(cls, mt_rest_obj):
...
-        auth_mode: str = ConnectionAuthMode.KEY,
-        **kwargs,
-    ):
-        configs = {"api_base": api_base, "api_type": api_type, "api_version": api_version, "auth_mode": auth_mode}
-        secrets = {"api_key": api_key} if auth_mode == ConnectionAuthMode.KEY else {}
-        self._token_provider = kwargs.get("token_provider")
-        super().__init__(configs=configs, secrets=secrets, **kwargs)
+
+class AzureOpenAIConnection(_CoreAzureOpenAIConnection, _StrongTypeConnection):
+    __doc__ = _CoreAzureOpenAIConnection.__doc__
```
Comment (wangchao1230) [inline src/promptflow/promptflow/_sdk/entities/_connection.py:284]: @crazygao please take a note on this

File: src/promptflow/promptflow/_sdk/entities/_connection.py (hunk 1 above)
Comment (brynn-code) [reply src/promptflow/promptflow/_sdk/entities/_connection.py:236]: Yes.

File: src/promptflow/promptflow/_sdk/entities/_connection.py (hunk 2 above)
Comment (crazygao) [reply src/promptflow/promptflow/_sdk/entities/_connection.py:284]: OK

**Review by crazygao — APPROVED** [review]

**Pull request merged**